### PR TITLE
Map 113 secure api key

### PR DESCRIPTION
**Pull request by jwillmartin**
# PR Details
## Description

Re-introduced private-resources to allow users to generate their own api keys to their deployments. In private-resources, replaced STOL's API keys with empty strings. Additionally, removed front-end links to private-resources and made it only accessible in the back-end. 

As keys will still be available in the network history, updated documentation to include guidance on securing Bing Maps API keys. Some relative links to future documentation were added in guidance. New documentation will be completed and links updated separately. 

## Related Jira Key

MAP-113

## Motivation and Context

Currently, the Bing Maps API keys are easily found by any user of the MAP Tool. These changes will remove them from the front end and will allow users to add their own API keys if they would like to deploy their own version of the MAP Tool. 

## How Has This Been Tested?
Built and tested locally using personally created api keys. 

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: private-resources/META-INF/maven/gov.usdot/fedgov-cv-private-resources-webapp/pom.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Generated by Apache Maven
+#Thu Dec 21 12:25:17 EST 2017
+version=1.0.0-SNAPSHOT
```
Comment (paulbourelly999) [inline private-resources/META-INF/maven/gov.usdot/fedgov-cv-private-resources-webapp/pom.properties:3]: Nit: the red `^M` text indicates windows line ending characters. When generating new files sometimes VSCode defaults to this line ending. We try to make all our files use linux line ending characters. To change it for a single file there should be a small option at the bottom of the VSCode editor when you have a file open:
image
If you could just update the couple of files that seem to have incorrect line endings that would be great. Again this is probably non critical so I will leave it up to you.

File: private-resources/WEB-INF/lib/cas-client-core-3.1.10.jar
Comment (paulbourelly999) [inline private-resources/WEB-INF/lib/cas-client-core-3.1.10.jar:1]: I don't think these dependencies need to be committed. I could be wrong but I imagine maven pulls them in during the build

File: private-resources/WEB-INF/web.xml
Comment (paulbourelly999) [inline private-resources/WEB-INF/web.xml:1]: What is the purpose of this file. What uses this file? Just curious

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
I thought this story was to address this 
https://github.com/usdot-fhwa-stol/connectedvcs-tools/blob/5d00c639894086b13cad97551c3c1467d78e4067/fedgov-cv-TIMcreator-webapp/src/main/webapp/index.html#L837
Is there any way we can do the same for the google API keys in the future? Am I missing something here?

The only reason I mention this at all is because the issue mentions both bing and google api keys. Since these keys have to be exposed on the client side anyway, the benefit of obscuring them here may be marginal. I am also ok with tracking change to the google API keys under a separate story later. Just wanted to clarify if that is meant to be included.

Additionally there seems to be a lot of duplication in the current obscuring of bing keys. Is there anyway we can define these functions in a single shared javascript file and it instead of having 4 copies of the same logic?

File: private-resources/META-INF/maven/gov.usdot/fedgov-cv-private-resources-webapp/pom.properties (hunk 1 above)
Comment (jwillmartin) [reply private-resources/META-INF/maven/gov.usdot/fedgov-cv-private-resources-webapp/pom.properties]: File was not needed, removed.

File: private-resources/WEB-INF/lib/cas-client-core-3.1.10.jar
Comment (jwillmartin) [reply private-resources/WEB-INF/lib/cas-client-core-3.1.10.jar:1]: We weren't building any of these, just zipping up the file. These dependencies weren't needed either, so I removed everything except the key files. Tested and works fine.

File: private-resources/WEB-INF/web.xml
Comment (jwillmartin) [reply private-resources/WEB-INF/web.xml:1]: It was used to point to the index.html, but that was empty and these aren't used. Only things needed in this folder were the keys, so I removed everything but the two key files.

**Review by paulbourelly999 — APPROVED** [review]
Looks good. As long as this is tested I am OK with the changeset. For the record it does include some technical debt (duplication) that we may want to address in the future.

**Review by DokurOmkar — CHANGES_REQUESTED** [review]

File: docs/BingMaps_API_Key_Guidance.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+In case the tool will not be publically available, the security feature does not need to be enabled. However, it is still recommended to periodically check the usage of your keys. 
+
+### **Getting a Bing Maps API Key**
+
+An account with Bing Maps can be created using the [Bing Maps Portal](https://www.bingmapsportal.com/). After an account is created, you may follow the official instructions on [Creating a Bing Maps Key](https://learn.microsoft.com/en-us/bingmaps/getting-started/bing-maps-dev-center-help/getting-a-bing-maps-key).
+
+### **Using your API Key**
+
+Once a key is obtained, please insert your key in the [ISDcreator-webapp-keys](/private-resources/js/ISDcreator-webapp-keys.js) file and build your updated ConnectedVCS Tools using either the [Local](/docs/) or [Docker](/docs/) build instructions.
```
[the remark this answers is not in the document]
Comment (jwillmartin) [reply docs/BingMaps_API_Key_Guidance.md]: They will be

File: private-resources/index.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (jwillmartin) [reply private-resources/index.html]: File along with others removed in previous commit

File: private-resources/js/ISDcreator-webapp-keys.js [hunk 4]
```diff
@@ -0,0 +1 @@
+var apiKey = "<bingMaps-isd-api-key>";
```
[the remark this answers is not in the document]
Comment (jwillmartin) [reply private-resources/js/ISDcreator-webapp-keys.js]: Yeah, that was also added back in the previous commit

File: fedgov-cv-TIMcreator-webapp/src/main/webapp/js/mapping.js [hunk 5]
```diff
@@ -37,12 +37,34 @@
         4.777314267158508, 2.388657133579254, 1.194328566789627,
         0.5971642833948135, 0.29858214169740677];
```
[the remark this answers is not in the document]
Comment (jwillmartin) [reply fedgov-cv-TIMcreator-webapp/src/main/webapp/js/mapping.js]: Fixed in latest commit

**Review by DokurOmkar — APPROVED** [review]

**Pull request merged**